ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCase.java

```diff
@@ -295,4 +314,208 @@ private static String standardizeTopology(final String topology) {
     return result.toString();
   }
 
+  public void createInputTopics(final FakeKafkaService fakeKafkaService) {
+    topics.forEach(fakeKafkaService::createTopic);
+  }
+
+  public void writeInputIntoTopics(
+      final FakeKafkaService fakeKafkaService,
+      final SchemaRegistryClient schemaRegistryClient
+  ) {
+    inputRecords.forEach(
+        record -> fakeKafkaService.writeSingleRecoredIntoTopic(
+            record.topic.getName(),
+            new FakeKafkaRecord(record, null))
+    );
+  }
+
+  @SuppressWarnings("unchecked")
+  public static void processInputFromTopic(
+      final TopologyTestDriverContainer testDriver,
+      final FakeKafkaService fakeKafkaService,
+      final KafkaTopicClient kafkaTopicClient,
+      final SchemaRegistryClient schemaRegistryClient
+  ) {
+    final List<FakeKafkaRecord> inputRecordsFromKafka = new ArrayList<>();
```

Comment:
I don't think we need this intermediate list (it just makes things harder to read), why not:
```java
for (KsqlTopic topic : testDriver.getSourceKsqlTopics()) {
  for (FakeKafkaRecord record : fakeKafkaService.readRecordsFromTopic(topic.getKafkaTopicName())) {
     processSingleRecord(record)
  }
}
```

Also as a nit, another reason I prefer `for foo : foos` instead of `.forEach` is that it makes the types easier to reason about (plus my usual gripes, like annoying stack traces and debugging)

Reply:
Agree, it's much easier to read and debug a for loop ;)